modules/generator/instance.go

```diff
@@ -151,8 +152,11 @@ func (i *instance) updateSubprocessors(desiredProcessors map[string]struct{}, de
 	_, latencyOk := desiredProcessors[spanmetrics.Latency.String()]
 	_, sizeOk := desiredProcessors[spanmetrics.Size.String()]
 
+	newDesiredProcessors := map[string]struct{}{}
+	maps.Copy(newDesiredProcessors, desiredProcessors)
```

Comment:
if desiredProcessors was capable of racing in the previous code, I don't see how this fixes it. it could still race here while copying to newDesiredProcessors

Reply:
This only reads the map, afaik concurrent map reads are fine. Later in this function we would modify `desiredProcessors` but with this change we only modify the copy `newDesiredProcessors`.